src/main/java/org/codebusters/audiogeek/spotifygazer/domain/spotify/connection/model/SpotifyNewReleasesAlbum.java

```diff
@@ -50,4 +57,11 @@ public Optional<Album> convertToAlbum() {
             return Optional.empty();
         }
     }
+
+    private LocalDate convertReleaseDateToLocalDate(String releaseDate) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(RELEASE_DATE_FORMAT);
```

Comment:
The `DateTimeFormatter` class implementation is immutable. See the javadocs:
```
* @implSpec
 * This class is immutable and thread-safe.
 *
 * @since 1.8
 */
public final class DateTimeFormatter {
```

Therefore there is no need to instantiate this object every time when do you need to convert date. You can safely extract it to private static constant. That would improve performance